src/sentry/identity/oauth2.py

```diff
@@ -152,70 +154,38 @@ def get_oauth_data(self, payload):
-
-        if req.status_code == 401:
-            logger.info(
-                "identity.oauth.refresh.identity-not-valid-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
-            )
-            raise IdentityNotValid(formatted_error)
-
-        if req.status_code == 400:
-            # this may not be common, but at the very least Google will return
-            # an invalid grant when a user is suspended
-            if error_name == "invalid_grant":
-                logger.info(
-                    "identity.oauth.refresh.identity-not-valid-error",
-                    extra={
-                        "error_name": error_name,
-                        "error_status_code": req.status_code,
-                        "error_description": error_description,
-                        "provider_key": self.key,
-                    },
-                )
-                raise IdentityNotValid(formatted_error)
-
-        if req.status_code != 200:
-            logger.info(
-                "identity.oauth.refresh.api-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
+    def get_refresh_token(
+        self, refresh_token, url: str, identity: Identity | RpcIdentity, **kwargs: Any
+    ) -> Response:
+        data = self.get_refresh_token_params(refresh_token, identity, **kwargs)
+
+        try:
+            req = safe_urlopen(
+                url=url,
+                headers=self.get_refresh_token_headers(),
+                data=data,
             )
-            raise ApiError(formatted_error)
+            req.raise_for_status()
+        except HTTPError as e:
+            error_resp = e.response
+            exc = ApiError.from_response(error_resp, url=url)
+            if isinstance(exc, ApiUnauthorized) or isinstance(exc, ApiInvalidRequestError):
+                raise IdentityNotValid from e
+            raise exc from e
+
+        return req
 
     def refresh_identity(self, identity: Identity | RpcIdentity, **kwargs: Any) -> None:
         refresh_token = identity.data.get("refresh_token")
 
         if not refresh_token:
             raise IdentityNotValid("Missing refresh token")
 
-        data = self.get_refresh_token_params(refresh_token, identity, **kwargs)
-
-        req = safe_urlopen(
-            url=self.get_refresh_token_url(), headers=self.get_refresh_token_headers(), data=data
+        req = self.get_refresh_token(
+            refresh_token=refresh_token,
+            url=self.get_refresh_token_url(),
+            identity=identity,
+            **kwargs,
```

Comment:
Any clue what the original `**kwargs` are used for in `get_refresh_token_params`?